### ISSUE TITLE
Integrate CI with the repository that runs the unit test.

### ISSUE THREAD

**Issue by DamianRodziewicz** [PredictiveEcology/SpaDES.shiny#28]

**Comment by DamianRodziewicz** [comment]
@achubaty @eliotmcintire - I see that this repository is configured with Travis (for linux and mac) and appveyor (for windows). Travis executes 4 builds (linux/R:devel, linux/R:oldrel, linux/R:release, osx). Each build builds the package and runs tests. One of the builds linters the code.

The current process sounds reasonable to me. The question is if there is anything that you'd like to change or upgrade in this process - i.e. would you like us to look into it and suggest improvements or should we just hook up `develop` branch?

**Comment by achubaty** [comment]
This configuration has worked well for us, but if you have additional suggestions, feel free to suggest. One thing that may become useful is using [`shinyTest`](https://github.com/rstudio/shinytest).

**Comment by achubaty** [comment]
I see that the suggested package `Rserve` is causing problems on travis-ci and possibly holding up #37 :

```r
Installing Rserve
'/home/travis/R-bin/lib/R/bin/R' --no-site-file --no-environ --no-save  \
  --no-restore --quiet CMD INSTALL  \
  '/tmp/Rtmpm7JoYh/devtools451f33061163/Rserve'  \
  --library='/home/travis/R/Library' --install-tests 
* installing *source* package ‘Rserve’ ...
checking whether to compile the server... yes
configure: error: R was configured without --enable-R-shlib or --enable-R-static-lib
*** Rserve requires R (shared or static) library.                       ***
*** Please install R library or compile R with either --enable-R-shlib  ***
*** or --enable-R-static-lib support                                    ***
 Alternatively use --without-server if you wish to build only Rserve client.
ERROR: configuration failed for package ‘Rserve’
* removing ‘/home/travis/R/Library/Rserve’
Installation failed: Command failed (1)
```

I'm not sure the best way to work around this, but I'll have a look.

**Comment by achubaty** [comment]
Appveyor is currently failing to to a warning in the documentation (warnings are treated as errors):

```r
* checking for missing documentation entries ... WARNING
Undocumented code objects:
  'slider'
All user-level objects in a package should have documentation entries.
See chapter 'Writing R documentation files' in the 'Writing R
Extensions' manual.
```

**Comment by achubaty** [comment]
NOTE: at this point, `Rserve` is not a strict dependency (it's only suggested; and even then, it's not used in the code). I've attempted a workaround for installing `Rserve` on travis-ci (686bdeac974b05e3e248fb1b1c2d88bf6f7cb6bc), but if that fails we can remove `Rserve` from the dependencies altogether.

**Comment by achubaty** [comment]
I've just made some tweaks (on the `development` branch) to resolve various CI issues. Please update the `develop` branch accordingly.

- update `.Rbuildignore` to fix note re: top-level files [4502e49f86c30c6345ab0b6d1961e52025248669]
- rebuilt `roxygen2` documentation [9b43f2c4b52c3f119e90e00b629e0c6470dae58d]
- removed a workaround for an error in `DiagrammeR` 0.9.1, which has since been fixed in 0.9.2 [89e620b133dd7bd47ac56758661657a246bbed9a]
- fix `RServe` installation error on travis-ci [686bdeac974b05e3e248fb1b1c2d88bf6f7cb6bc]
- update minimum R version to 3.3.3 based on package dependencies (this solves appveyor failures on R 3.3.2) [fd6c16489db7e8ee8b9eda12903a2ae56728db74]

**Comment by DamianRodziewicz** [comment]
We'll merge this branch once CI passes successfully.
